Sources/DataPipeline/DataPipelineImplementation.swift

```diff
@@ -341,13 +341,17 @@ extension DataPipelineImplementation {
         analytics.process(event: trackRegisterTokenEvent)
     }
 
-    func processPushMetricsFromBGQ(token: String, event: Metric, deliveryId: String, timestamp: String, metaData: [String: Any] = [:]) {
-        let properties: [String: Any] = metaData.mergeWith([
-            "metric": event.rawValue,
-            "deliveryId": deliveryId,
-            "recipient": token
+    func processPushMetricsFromBGQ(token: String?, event: String, deliveryId: String, timestamp: String, metaData: [String: Any]) {
+        var properties: [String: Any] = metaData.mergeWith([
+            "metric": event,
+            "deliveryId": deliveryId
         ])
+        if let token = token {
+            properties["recipient"] = token
+        }
         var trackPushMetricEvent = TrackEvent(event: "Report Delivery Event", properties: try? JSON(properties))
+        // anonymousId or userId is required in the payload for backend processing
+        trackPushMetricEvent.anonymousId = deliveryId
```

Comment:
I am not confident in my CDP API knowledge to review this line of code. If someone else on the team could review this, I would appreciate it.

Reply:
CDP (backend) _requires_ setting of either `anonymousId` or `userId` in the request payload to process it. If either of these is missing then the request payload is not processed.